src/components/Controls/ArrayInput/hooks/index.ts

```diff
@@ -0,0 +1,43 @@
+import {useState, KeyboardEventHandler, ChangeEventHandler} from "react"
+import type {ArrayInputChangeListener, ArrayInputValue} from "../"
+
+type UseArrayInput = {
+	inputValue: string
+	handleInputChange: InputChangeListener
+	handleKeyDown: KeyboardEventHandler<HTMLDivElement>
+}
+
+type InputChangeListener = ChangeEventHandler<HTMLInputElement>
+
+const useArrayInput = ({
+	onChange,
+	value
+}: {
+	onChange: ArrayInputChangeListener
+	value: ArrayInputValue
+}): UseArrayInput => {
+	const [inputValue, setInputValue] = useState<string>("")
+
+	const handleInputChange: InputChangeListener = event => {
+		setInputValue(event.target.value)
+	}
+
+	const handleKeyDown: KeyboardEventHandler<HTMLDivElement> = event => {
+		if (!inputValue) return
+		switch (event.key) {
+			case "Enter":
+			case "Tab":
+				const newValue = [...value, inputValue]
```

Comment:
This is not consistent. You have onRemove(index) handler below, so, first, it's more obvious to call this handler onAdd instead of onChange, second, just call it with the added element instead of constructing the whole array here.
Imagine, we could do something with added element before updating the array in the parent component, and this way doesn't allow to see which element is added.